ow_lander/scripts/path_planning_commander.py

```diff
@@ -194,8 +205,29 @@ def run(self):
     self.guarded_move_srv = rospy.Service(
         'arm/guarded_move', GuardedMove, self.handle_guarded_move)
     print("path_planning_commander has started!")
+
     rospy.spin()
 
+  def stop_arm(self):
+    plan = activity_guarded_move.stop_guarded_move_plan(self.arm_move_group)
+    self.trajectory_async_executer.systemFaultListener(plan.joint_trajectory, self.handle_guarded_move_feedback)
+
+  def callback(self, data):
+    """
+    If system fault occurs, and it is an arm failure, path planning stops
+    """
+    if data.value == 4:
+      self.arm_fault = True
+      self.trajectory_async_executer.stop()
+    else:
+      self.arm_fault = False
```

Comment:
I think the logic around when and for how long a fault is introduced needs some discussion, I'll post some notes in my review summary.